File: backend/src/models/pipeline/DeployTimes.ts
export class DeployInfo {
  pipelineCreateTime: string;
  jobStartTime: string;
  jobFinishTime: string;
  commitId: string;
  state: string;

  constructor(
    pipelineCreateTime: string,
    jobStartTime: string,
    jobFinishTime: string,
    commitId: string,
    state: string
  ) {
    this.pipelineCreateTime = pipelineCreateTime;
    this.jobStartTime = jobStartTime;
    this.jobFinishTime = jobFinishTime;
    this.commitId = commitId;
    this.state = state;
  }
}

export class DeployTimes {
  pipelineId: string;
  pipelineName: string;
  pipelineStep: string;
  passed: DeployInfo[];
  failed: DeployInfo[];

  constructor(
    pipelineId: string,
    pipelineName: string,
    pipelineStep: string,
    passed: DeployInfo[],
    failed: DeployInfo[]
  ) {
    this.pipelineId = pipelineId;
    this.pipelineName = pipelineName;
    this.pipelineStep = pipelineStep;
    this.passed = passed;
    this.failed = failed;
  }
}

File: backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/DeployTimes.java
package heartbeat.client.dto.pipeline.buildkite;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DeployTimes {

	private String pipelineId;

	private String pipelineName;

	private String pipelineStep;

	private List<DeployInfo> passed;

	private List<DeployInfo> failed;

}
